aws-geo-location/src/main/java/com/amplifyframework/geo/location/service/AmazonLocationService.kt

```diff
@@ -50,6 +57,48 @@ internal class AmazonLocationService(
         return readFromBuffer(response.blob)
     }
 
+    override fun geocode(
+        index: String,
+        query: String,
+        limit: Int,
+        area: SearchArea?,
+        countries: List<CountryCode>
+    ): List<Place> {
+        val request = SearchPlaceIndexForTextRequest()
+            .withIndexName(index)
+            .withText(query)
+            .withMaxResults(limit)
+            .withFilterCountries(countries.map { it.name })
+        if (area?.biasPosition != null) {
+            val position = listOf(
+                area.biasPosition!!.longitude,
+                area.biasPosition!!.latitude
+            )
+            request.setBiasPosition(position)
+        } else if (area?.boundingBox != null) {
+            val boundary = listOf(
+                area.boundingBox!!.longitudeSW,
+                area.boundingBox!!.latitudeSW,
+                area.boundingBox!!.longitudeNE,
+                area.boundingBox!!.latitudeNE
+            )
+            request.setFilterBBox(boundary)
+        }
```

Comment:
You could avoid the not-null assert operator here, one option would be using `let {}`:

```suggestion
        area?.biasPosition?.let { request.setBiasPosition(listOf(it.longitude, it.latitude)) }
        area?.boundingBox?.let {
            request.setFilterBBox(listOf(it.longitudeSW, it.latitudeSW, it.longitudeNE, it.latitudeNE))
        }
```

Reply:
huge fan of this. very kotlin-esque